pkg/logging/scalyr/api_test.go

```diff
@@ -0,0 +1,433 @@
+package scalyr_test
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/fastly/cli/pkg/app"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/mock"
+	"github.com/fastly/cli/pkg/testutil"
+	"github.com/fastly/cli/pkg/update"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+func TestScalyrCreate(t *testing.T) {
+	for _, testcase := range []struct {
+		args       []string
+		api        mock.API
+		want       *fastly.Scalyr
+		wantError  string
+		wantOutput string
+	}{
+		{
+			args:      []string{"logging", "scalyr", "create", "--service-id", "123", "--version", "1", "--name", "", "--auth-token", "abc"},
```

Comment:
@phamann I added this test case as explicitly-set zero values in cli args results in runtime panics, hence the validation at the CLI layer.

Maybe runtime panics for this behavior are acceptable over validation. I do think this is an edge case as it appropriately and nicely handles omitted required flags.